Applications/Spire/Source/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.cpp

```diff
@@ -1 +1,57 @@
 #include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp"
+#include "Spire/Spire/Utility.hpp"
+
+using namespace Nexus;
+using namespace Spire;
+
+void LocalOrderImbalanceIndicatorModel::publish(
+    const OrderImbalance& imbalance) {
+  m_last_published_imbalance = imbalance;
+  insert_sorted(imbalance);
+  m_imbalance_published_signal(imbalance);
+}
+
+void LocalOrderImbalanceIndicatorModel::insert(
+    const OrderImbalance& imbalance) {
+  insert_sorted(imbalance);
+}
+
+QtPromise<std::vector<Nexus::OrderImbalance>>
+    LocalOrderImbalanceIndicatorModel::load(const TimeInterval& interval) {
+  auto first = std::lower_bound(m_imbalances.begin(), m_imbalances.end(),
+    interval.lower(), [] (const auto& imbalance, const auto& timestamp) {
+      return imbalance.m_timestamp < timestamp;
+    });
+  auto last = std::upper_bound(m_imbalances.begin(), m_imbalances.end(),
+    interval.upper(), [] (const auto& timestamp, const auto& imbalance) {
+      return imbalance.m_timestamp > timestamp;
+    });
+  return QtPromise(
+    [imbalances = std::vector<OrderImbalance>(first, last)] () mutable {
+      return std::move(imbalances);
+  });
+}
+
+SubscriptionResult<boost::optional<Nexus::OrderImbalance>>
+    LocalOrderImbalanceIndicatorModel::subscribe(
+    const OrderImbalanceSignal::slot_type& slot) {
+  return {m_imbalance_published_signal.connect(slot),
+    QtPromise([imbalance = m_last_published_imbalance] () mutable {
```

Comment:
Use m_imbalances.back() as the last published imbalance. If it doesn't exist then use ```boost::none```.

Reply:
How can we make that assumption, though? If I insert A and C, then publish B, the subscribe method will return C as the last published imbalance. Or is the last published imbalance strictly the newest imbalance?

Reply:
Good point, the intention is that the ```subscribe``` method returns the most recent imbalance. ```insert``` is used to add historical imbalances and ```publish``` is used to add the newest imbalance.